versioned/gc/core/src/main/java/org/projectnessie/versioned/gc/IdentifyUnreferencedAssets.java

```diff
@@ -143,17 +114,21 @@ public boolean call(Row r) throws Exception {
 
     private boolean referenced;
     private byte[] data;
+    private long hashedData;
```

Comment:
I think it is a bad idea to do pre-hashing. It seems like it is likely to substantially increase the bloomfilter false positives. Instead, I propose just using the inner contents field of bytes instead of the outer object.

Reply:
I dont love the pre-hashing either. I am not sure I understand your suggestion though. Do you suggest to use the byte bloom filter and have two byte representations of the asset key? Or using a better hashing algo?

My thinking: an AssetKey has a set of 'unique' fields and a set of 'informational' fields. Serialising this to bytes and using the byte array bloom filter means all the fields are considered so in the iceberg case the same data file referenced from different snapshots (or tables) is treated as 2 different Assets. I had understood your forcing `equals` and `hashCode` to be implemented by `AssetKey`s as a recognition of this and the most reliable way to compare objects.

Reply:
My suggestion was based on thinking this was at the value level, not the asset key level. I think the "right" thing to do is probably to make an asset key be composed of two binary values, the unique parts and the non-unique parts. And/or make it so that the reportable name is well defined e.g.:

AssetKey {
  repeated string name;
  byte[] data;
}

That way we can add additional generic metadata fields while declaring they aren't considered for uniqueness.

Reply:
I have taken the 2nd option by exposing a unique name. The two byte string representation felt like a heavy change. I didn't use the reportable name as that to me contains the extra metadata, instead I exposed a unique name, I don't love this though, perhaps the reportable name should be the unique one and the toString shoudl show metadata?